### PR TITLE
Add PA mulit-model test

### PR DESCRIPTION
**Pull request by matthewkotila**

### CODE REVIEW

**Review by tanmayv25 — APPROVED** [review]

**Comment by matthewkotila** [comment]
@tanmayv25 I'll wait until CI is working again to see that the test added from this PR passes before merging this PR 👍

**Comment by tanmayv25** [comment]
@matthewkotila  Is there a way in MPI to also check the timelines of the processes? With your changes all these processes should end almost at the same time.

**Comment by tanmayv25** [comment]
One more thing, can you also add another PA invocation with a custom model with large execution delay? This in conjunction with the above ask should provide a strong test case.

**Comment by matthewkotila** [comment]
> @matthewkotila Is there a way in MPI to also check the timelines of the processes? With your changes all these processes should end almost at the same time.

What's the purpose of this, though? To test that a feature of the MPI library works correctly? If so, that seems unnecessary.

**Comment by matthewkotila** [comment]
> One more thing, can you also add another PA invocation with a custom model with large execution delay? This in conjunction with the above ask should provide a strong test case.

Sounds like a good idea--do you have an example of a model/some simple steps I can follow to test this out?

**Comment by tanmayv25** [comment]
You can see here:
https://github.com/triton-inference-server/server/blob/main/qa/L0_long_running_stress/test.sh#L115-L120

Currently PR tests that all the models have individually statbilized, but we need to test that the process with short running model doesn't exits when the process with the long model is still running.

**Review by tanmayv25 — APPROVED** [review]

**Comment by matthewkotila** [comment]
@tanmayv25 I cleaned up some stuff in the test--please re-review/approve 🙏

**Review by tanmayv25 — APPROVED** [review]

File: qa/L0_perf_analyzer/test.sh
Comment (jbkyang-nvi) [inline qa/L0_perf_analyzer/test.sh:777]: What is this deviation calculation from? Can you add a comment for the source?

Comment (jbkyang-nvi) [inline qa/L0_perf_analyzer/test.sh:777]: Is this a common devation calculation? Can you maybe link an explaination of where this is from?

Comment (matthewkotila) [reply qa/L0_perf_analyzer/test.sh:777]: it's based on the equivalent thresholding logic in the perf analyzer source code:

https://github.com/triton-inference-server/client/blob/matthewkotila-mpi/src/c++/perf_analyzer/inference_profiler.cc#L629-L644

Comment (matthewkotila) [reply qa/L0_perf_analyzer/test.sh:777]: it just is calculating whether each throughout/latency from the last three throughputs/latencies is more than X percent different from the mean of the last three throughputs/latencies

Comment (jbkyang-nvi) [reply qa/L0_perf_analyzer/test.sh:777]: I see. I would prefer if inference_profiler code was linked here for readability

Comment (matthewkotila) [reply qa/L0_perf_analyzer/test.sh:777]: i assume by "here", you mean the test source code, not this PR conversation. adding comment to test source code... 👍

Comment (matthewkotila) [reply qa/L0_perf_analyzer/test.sh:778]: added comment 👍

**Review by jbkyang-nvi — DISMISSED** [review]

**Comment by matthewkotila** [comment]
@jbkyang-nvi I rebased--please re-approve. Also if your conversation from above is resolved, can you resolve it?

**Review by jbkyang-nvi — DISMISSED** [review]

**Comment by matthewkotila** [comment]
@tanmayv25 

Added MPI installation to test container for this new test--need review/approval 👍🙏

**Review by tanmayv25 — APPROVED** [review]

File: Dockerfile.QA
Comment (tanmayv25) [inline Dockerfile.QA:281]: Installing this fixed the issue?

**Review by tanmayv25 — APPROVED** [review]

File: Dockerfile.QA
Comment (matthewkotila) [reply Dockerfile.QA:281]: Yes 👍

**Review by tanmayv25 — DISMISSED** [review]

File: qa/L0_perf_analyzer/test.sh
Comment (tanmayv25) [inline qa/L0_perf_analyzer/test.sh:123]: What happens without this commit?

Comment (matthewkotila) [reply qa/L0_perf_analyzer/test.sh]: We switched to the LD_LIBRARY_PATH solution 👍

File: Dockerfile.QA
Comment (tanmayv25) [inline Dockerfile.QA:333]: Add a comment with the DLIS ticket id over this line for why it is needed.

File: Dockerfile.sdk
Comment (tanmayv25) [inline Dockerfile.sdk:245]: Add a comment with the DLIS ticket Misha is working on over this line.

**Review by tanmayv25 — COMMENTED** [review]
Minor comments. otherwise LGTM.

File: Dockerfile.QA
Comment (matthewkotila) [reply Dockerfile.QA:346]: Added comment 👍

File: Dockerfile.sdk
Comment (matthewkotila) [reply Dockerfile.sdk:247]: Added comment 👍

**Review by tanmayv25 — APPROVED** [review]

**Pull request merged**
